src/XMakeCommandLine/XMake.cs

```diff
@@ -226,6 +226,21 @@ private static void HandleConfigurationException(Exception ex)
         }
 
         /// <summary>
+        /// Insert the command executable path as the first element of the args array.
+        /// </summary>
+        /// <param name="args"></param>
+        /// <returns></returns>
+        private static string[] ConstructArrayArg(string[] args)
```

Comment:
@AndyGerlicher @dsplaisted 

QuotingUtilities.SplitUnquoted is called in MSBuildApp.GatherAllSwitches to break down the command line string into an argument array. By the looks of it, it splits on white space when no separators are provided.